src/methods/call.js

```diff
@@ -316,72 +317,75 @@ export function OrdinaryCallEvaluateBody(
         throw new FatalError(err);
       }
     }
-  } else if (F.$FunctionKind === "generator") {
-    // 1. Perform ? FunctionDeclarationInstantiation(functionObject, argumentsList).
-    FunctionDeclarationInstantiation(realm, F, argumentsList);
-
-    // 2. Let G be ? OrdinaryCreateFromConstructor(functionObject, "%GeneratorPrototype%", « [[GeneratorState]], [[GeneratorContext]] »).
-    let G = OrdinaryCreateFromConstructor(realm, F, "GeneratorPrototype", {
-      $GeneratorState: undefined,
-      $GeneratorContext: undefined,
-    });
+  } else {
+    invariant(F instanceof ECMAScriptFunctionValue);
```

Comment:
Or even better, it sounds like the type of F in the function signature should be `NativeFunctionValue | ECMAScriptFunctionValue`, excluding `BoundFunctionValue`.